File: examples/interactive_calculator.py
# Interactive Calculator Example
# This demonstrates how the interactive input/output works

def calculator():
    print("Welcome to the Interactive Calculator!")
    print("Enter 'q' to quit at any time.")
    
    while True:
        expression = input("Enter an expression (e.g., 2 + 3): ")
        
        if expression.lower() == 'q':
            print("Thank you for using the Interactive Calculator!")
            break
        
        try:
            # Safely evaluate the expression
            result = eval(expression)
            print(f"Result: {result}")
        except Exception as e:
            print(f"Error: {str(e)}")
            print("Please try again with a valid expression.")

# Run the calculator
calculator()

File: examples/interactive_calculator.js
// Interactive Calculator Example
// This demonstrates how the interactive input/output works

const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function calculator() {
  console.log("Welcome to the Interactive Calculator!");
  console.log("Enter 'q' to quit at any time.");
  
  function promptUser() {
    rl.question("Enter an expression (e.g., 2 + 3): ", (expression) => {
      if (expression.toLowerCase() === 'q') {
        console.log("Thank you for using the Interactive Calculator!");
        rl.close();
        return;
      }
      
      try {
        // Safely evaluate the expression
        const result = eval(expression);
        console.log(`Result: ${result}`);
      } catch (e) {
        console.log(`Error: ${e.message}`);
        console.log("Please try again with a valid expression.");
      }
      
      // Continue prompting
      promptUser();
    });
  }
  
  // Start the prompt loop
  promptUser();
}

// Run the calculator
calculator();
